profile.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package eat
+
+import (
+	"strconv"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+	"encoding/asn1"
+	cbor "github.com/fxamacker/cbor/v2"
+)
+
+// Profile is either a well formed RFC3986 compliant URI(uri  *url.URL).
+// or an ObjectIdentifier  ([]int)
+type Profile struct {
+	val interface{}
+}
+
+// DecodeProfileCBOR decodes from a received CBOR data the profile
+// as either a URL or a Object Identifier
+func (s Profile) DecodeProfileCBOR(val interface{}) error
+{
+	switch t := val.(type) {
+	case string:
+		if len(val) == 0 {
+			return fmt.Errorf("No valid URL for profile")
+		}
+		s.val, err := url.Parse(s.val)
```

Comment:
```suggestion
		s.val, err := url.Parse(t)
```

Reply:
Corrected and removed t variable altogether!

Reply:
I think we should keep `t` instead